cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/web/ACSearchRunner.java

```diff
@@ -287,23 +267,88 @@ private void search(Subject subject)
         }
     }
 
-    private void writeError(SyncOutput syncOutput, int code, Throwable t)
-    {
-        try
-        {
-            syncOutput.setResponseCode(code);
+    private void writeError(SyncOutput syncOutput, int code, Throwable t) {
+        try {
+            syncOutput.setCode(code);
             syncOut.setHeader("Content-Type", "text/plain");
             OutputStream ostream = syncOut.getOutputStream();
-            if (ostream != null)
-            {
+            if (ostream != null) {
                 OutputStreamWriter w = new OutputStreamWriter(ostream);
                 w.write(t.toString());
                 w.flush();
             }
-        }
-        catch (IOException e)
-        {
+        } catch (IOException e) {
             log.warn("Could not write response to output stream", e);
         }
     }
+
+    /**
+     * Validate the incoming parameters.
+     * 
+     * IVOA GMS requests can be:
+     *   - (no params) - list group memberships
+     *   - group=[groupName] - is member of {groupName}?
+     * CADC legacy requests can be:
+     *   - role=[role] - list groups in role (member, admin, or owner)
+     *   - role=[role] groupID=[groupName}] - list group if user has specified role in group
+     * 
+     * @param params The incoming parameters.
+     */
+    protected void validateParams(List<Parameter> params) {
+        
+        if (params == null) {
+            throw new IllegalArgumentException("missing parameters");
+        }
+        
+        String roleParam = ParameterUtil.findParameterValue("ROLE", params);
+        String groupIDParam = ParameterUtil.findParameterValue("GROUPID", params);
+        // ivoa param name is 'group'
+        String groupParam = ParameterUtil.findParameterValue("group", params);
+        
+        // ivoa search for all memberships is request with no params
+        if (roleParam == null && groupIDParam == null && groupParam == null) {
+            // default to membership role
+            this.role = Role.MEMBER;
+            ivoaStandardResponse = true;
+            return;
+        }
+        
+        if (groupParam != null) {
+            this.groupName = groupParam;
+            if (roleParam != null) {
+                // allow role with vo param, but use old xml response
+                this.role = Role.toValue(roleParam);
+            } else {
+                ivoaStandardResponse = true;
+                role = Role.MEMBER;
+            }
+            return;
+        }
+        
+        // role is required
+        if (roleParam != null) {
+            this.role = Role.toValue(roleParam);
+        } else {
+            throw new IllegalArgumentException("required parameter ROLE not found");
+        }
+        
+        // groupID is optional
+        if (groupIDParam != null) {
+            this.groupName = groupIDParam;
+        }
+
+    }
+    
+    protected Role getRole() {
```

Comment:
Don't see any use for these methods. If test code it should be package access, not protected.

Reply:
Yes, for test code.  Changed to package level.